src/app/hooks/use-push-notifications.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+
+import { initializeApp } from 'firebase/app';
+import { getMessaging, getToken } from 'firebase/messaging';
+import browser from 'webextension-polyfill';
+
+import { AnalyticsEventCategory, useAnalytics } from 'lib/analytics';
+import { EnvVars } from 'lib/env';
+import { useLocalStorage } from 'lib/ui/local-storage';
+
+export const usePushNotifications = () => {
+  const [fcmToken, setFcmToken] = useLocalStorage<string | undefined>('fcmToken', undefined);
+  const { trackEvent } = useAnalytics();
+
+  useEffect(() => {
+    if ('serviceWorker' in navigator && 'Notification' in window) {
+      initializeApp(JSON.parse(EnvVars.TEMPLE_FIREBASE_CONFIG));
+      const messaging = getMessaging();
+
+      Notification.requestPermission().then(async permission => {
+        if (permission === 'granted') {
+          if (!fcmToken) {
```

Comment:
Looks like it is possible to remove nested conditionals and use the same trick as above